native/packages/react-native-bpk-component-icon/src/BpkIcon.js

```diff
@@ -0,0 +1,83 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2017 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Text, Platform, StyleSheet, View } from 'react-native';
+import iconMappings from 'bpk-svgs/dist/font/iconMapping.json';
+
+const tokens = Platform.select({
+  ios: () => require('bpk-tokens/tokens/ios/base.react.native.common.js'), // eslint-disable-line global-require
+  android: () => require('bpk-tokens/tokens/android/base.react.native.common.js'), // eslint-disable-line global-require
+})();
+
+const iconNamePropType = (props, propName, componentName) => {
+  const iconName = props[propName];
+  if (!iconMappings[iconName]) {
+    return new Error(`Invalid prop \`${propName}\` supplied to \`${componentName}\`. No icon matches the name \`${iconName}\`.`); // eslint-disable-line max-len
+  }
+  return false;
+};
+
+const {
+  spacingBase,
+  spacingLg,
+} = tokens;
+
+const styles = StyleSheet.create({
+  icon: {
+    width: spacingLg,
```

Comment:
Why is width being set here? (not saying it's wrong, im just trying to understand these styles)

Reply:
If we don't set width, then the text component will automatically be "1 character" wide, which doesn't work very well with wider icons

![screen shot 2017-09-29 at 16 28 42](https://user-images.githubusercontent.com/30267516/31023323-6566316e-a533-11e7-8b7e-bc827489a602.png)